src/Nordea/Components/TextInput.elm

```diff
@@ -92,16 +111,36 @@ withError condition (TextInput config) =
     TextInput { config | showError = condition }
 
 
+withSearchIcon : Bool -> TextInput msg -> TextInput msg
+withSearchIcon condition (TextInput config) =
+    TextInput { config | hasSearchIcon = condition }
+
+
 
 -- VIEW
 
 
 view : List (Attribute msg) -> TextInput msg -> Html msg
 view attributes (TextInput config) =
-    styled input
-        (getStyles config)
-        (getAttributes config ++ attributes)
-        []
+    Html.div [ css [ position relative ] ]
+        [ Html.span
+            [ css
+                [ position absolute
+                , left (rem 0.7)
+                , top (pct 50)
+                , transform (translateY (pct -50))
+                , displayFlex
+                , alignItems center
+                , pointerEvents none
+                ]
+            ]
+            [ Icons.search [ css [ width (rem 1.4), opacity (num 0.5) ] ] ]
+            |> showIf config.hasSearchIcon
+        , styled input
+            (getStyles config)
+            (getAttributes config ++ attributes)
```

Comment:
I think the attributes should be added to the root element, in this case the wrapping div, to better support adding margins etc.

```suggestion
            (getAttributes config)
```